.github/workflows/main.yml

```diff
@@ -0,0 +1,69 @@
+---
+name: Tests
+
+on:
+  pull_request: ~
+  push:
+    branches: [ master ]
+
+  # Allow job to be triggered manually.
+  workflow_dispatch:
+
+# Cancel in-progress jobs when pushing to the same branch.
+concurrency:
+  cancel-in-progress: true
+  group: ${{ github.workflow }}-${{ github.ref }}
+
+jobs:
+
+  tests:
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["ubuntu-20.04"]
+        python-version: ["3.6", "3.11"]
```

Comment:
I've deliberately selected only the lower and upper bound of Python versions to be tested on, to save resources. Please let me know if you want this to be expanded to enumerate all Python versions.